pkg/service/dbaas/service.go

```diff
@@ -101,13 +104,25 @@ func (s *Service) Execute(ctx context.Context) error {
 	return p.RunWithContext(&Context{Context: ctx})
 }
 
-// fetchManagedDBaaS fetches instances from kubernetes cluster
-func (s *Service) fetchManagedDBaaS(ctx *Context) error {
+// fetchManagedDBaaSAndNamespaces fetches instances and namespaces from kubernetes cluster
+func (s *Service) fetchManagedDBaaSAndNamespaces(ctx *Context) error {
 	log := ctrl.LoggerFrom(ctx)
 
+	log.V(1).Info("Listing namespaces from cluster")
+	namespaces, err := fetchNamespaces(ctx, s.k8sClient)
+	if err != nil {
+		return fmt.Errorf("cannot list namespaces: %w", err)
+	}
+
+	exists, err := labels.NewRequirement(service.NamespaceLabel, selection.Exists, nil)
+	if err != nil {
+		return fmt.Errorf("cannot create label requirement: %w", err)
+	}
 	var dbaasDetails []Detail
 	for _, groupVersionResource := range groupVersionResources {
-		managedResources, err := s.k8sClient.Resource(groupVersionResource).List(ctx, metav1.ListOptions{})
+		managedResources, err := s.k8sClient.Resource(groupVersionResource).List(ctx, metav1.ListOptions{
+			LabelSelector: exists.String(),
+		})
```

Comment:
Nitpick: I don't think you really need to filter this. Shouldn't all resources have this namespace label? And even if not you double check that they actually have one a few lines down anyways